lib/DAV/Calendar.php

```diff
@@ -0,0 +1,199 @@
+				'protected' => true,
+			],
+			[
+				'privilege' => '{DAV:}read',
+				'principal' => $this->getOwner() . '/calendar-proxy-write',
+				'protected' => true,
+			],
+			[
+				'privilege' => '{DAV:}read',
+				'principal' => $this->getOwner() . '/calendar-proxy-read',
+				'protected' => true,
+			],
+		];
+		if ($this->backend->checkBoardPermission($this->board->getId(), Acl::PERMISSION_EDIT)) {
+			$acl[] = [
+				'privilege' => '{DAV:}write',
+				'principal' => $this->getOwner(),
+				'protected' => true,
+			];
+			$acl[] = [
+				'privilege' => '{DAV:}write-properties',
+				'principal' => $this->getOwner(),
+				'protected' => true,
+			];
+		}
+		return $acl;
+	}
+
+	public function setACL(array $acl) {
+		throw new Forbidden('Setting ACL is not supported on this node');
+	}
+
+	public function getSupportedPrivilegeSet() {
+		return null;
+	}
+
+	public function calendarQuery(array $filters) {
+		// FIXME: In a real implementation this should actually filter
+		return array_map(function ($card) {
+			return $card->getCalendarPrefix() . '-' . $card->getId() . '.ics';
+		}, $this->children);
+	}
+
+	public function createFile($name, $data = null) {
+		throw new \Sabre\DAV\Exception\Forbidden('Creating a new entry is not implemented');
+	}
+
+	public function getChild($name) {
+		if ($this->childExists($name)) {
+			$card = array_values(array_filter(
+				$this->children,
+				function ($card) use (&$name) {
+					return $card->getCalendarPrefix() . '-' . $card->getId() . '.ics' === $name;
+				}
+			));
+			if (count($card) > 0) {
+				return new CalendarObject($this, $name, $card[0], $this->backend);
+			}
+		}
+	}
+
+	public function getChildren() {
+		$childNames = array_map(function ($card) {
+			return $card->getCalendarPrefix() . '-' . $card->getId() . '.ics';
+		}, $this->children);
+
+		$children = [];
+
+		foreach ($childNames as $name) {
+			$children[] = $this->getChild($name);
+		}
+
+		return $children;
+	}
+
+	public function childExists($name) {
+		return count(array_filter(
+			$this->children,
+			function ($card) use (&$name) {
+				return $card->getCalendarPrefix() . '-' . $card->getId() . '.ics' === $name;
+			}
+		)) > 0;
+	}
+
+
+	public function delete() {
+		return null;
+	}
+
+	public function getLastModified() {
+		return $this->board->getLastModified();
+	}
+
+	public function getGroup() {
+		return [];
+	}
+
+	public function propPatch(PropPatch $propPatch) {
+		$properties = [
+			'{DAV:}displayname',
+			'{http://apple.com/ns/ical/}calendar-color'
+		];
+		$propPatch->handle($properties, function ($properties) {
+			foreach ($properties as $key => $value) {
+				switch ($key) {
+					case '{DAV:}displayname':
+						if (mb_substr($value, 0, strlen('Deck: '))) {
+							$value = mb_substr($value, strlen('Deck: '));
+						}
+						$this->board->setTitle($value);
+						break;
+					case '{http://apple.com/ns/ical/}calendar-color':
+						$this->board->setColor(substr($value, 1));
+						break;
```

Comment:
@georgehrke Is there any standard where the format of this is specified? Are there any other values than hex colors that could occur here?

Reply:
ping :wink:

Reply:
`{http://apple.com/ns/ical/}calendar-color` always contains a hex color. Usually it's `#rrggbb`. macOS calendar is sending `#rrggbbaa` though, so you would have to filter that out.